### DISCUSSION TITLE
Very slow page transition on `zagjs.com`

### DISCUSSION THREAD

**Discussion by Odas0R** [chakra-ui/zag#846] [Ideas]
When navigating through the components

![image](https://github.com/chakra-ui/zag/assets/32167770/d5ab3df5-9bf3-42d8-bd6c-c5f8d79a6958)

Takes about 3-4s to change to the other page. I imagine the docs is an SPA? Maybe someone found the same problem

**Comment by cschroeter** [comment]
wtf?! 🫠

**Comment by Odas0R** [comment]
> wtf?! 🫠

Ahahaha, a video explains it better, sure. Very rushed gh issue.

**In the video**:
I clicked "Context Menu" right at the beginning of the video, and it took about 4+ seconds.

Weirdly this doesn't happen at first (after a couple of different pages), then starts happening out of nowhere. This also happened to a colleague of mine.

[Screencast from 09-09-2023 07:59:25 PM.webm](https://github.com/chakra-ui/zag/assets/32167770/254db4a4-39d2-4e59-944f-ceaf75fd9cca)

I bet there is some weird, async component rendering happening here, and it hangs till it renders, maybe wrong, don't know.

This is a real issue, not a troll @cschroeter.

**Comment by cschroeter** [comment]
@Odas0R nono i could reproduce the issue. i have a educated guess why this may be the case @segunadebayo https://github.com/chakra-ui/zag/commit/4e2e589a890dba4467b84f878dcb272bc0b77ccd#diff-1bcb3c6408eb0349a1fdc6950830017372d6ebd94fa2068dc6045c4364772f4f

thanks for bringing this up

**Comment by cschroeter** [comment]
@Odas0R Issue should be solved by now

**Reply by Odas0R** [reply]
Yeah looks good :). Very nice :muscle:
